### PR TITLE
Add a test for the linear solver SPRAL

### PR DESCRIPTION
**Pull request by amontoison**
@odow 
Do you know how we can implement the following code without JuMP?
```julia
model = Model(Ipopt.Optimizer)
@variable(model, x)
@objective(model, Min, (x - 2)^2)

# Use the linear solver SPRAL
set_attribute(model, "linear_solver", "spral")
optimize!(model)
@test value(x) == 2.0
```

### CODE REVIEW

**Comment by chriscoey** [comment]
@amontoison try this (no need for JuMP dependency):
```
    model = Ipopt.Optimizer()
    MOI.set(model, MOI.RawOptimizerAttribute("linear_solver"), "spral")    
    MOI.set(model, MOI.Silent(), true)
    x = MOI.add_variable(model)
    MOI.set(model, MOI.ObjectiveSense(), MOI.MIN_SENSE)
    f = (1.0 * x - 2.0) ^ 2
    MOI.set(model, MOI.ObjectiveFunction{typeof(f)}(), f)
    MOI.optimize!(model)
    @test MOI.get(model, MOI.VariablePrimal(), x) ≈ 2.0
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,6 +23,11 @@ jobs:
             os: windows-latest
             arch: x86
     steps:
+      - name: Set environment variables for OpenMP
+        shell: bash
+        run: |
+          echo "OMP_CANCELLATION=TRUE" >> $GITHUB_ENV
+          echo "OMP_PROC_BIND=TRUE" >> $GITHUB_ENV
```
Comment (odow) [inline .github/workflows/ci.yml:30]: You should instead add a `env:` field under `users: Julia-actions/runtest@v1`

File: test/MOI_wrapper.jl [hunk 2]
```diff
@@ -476,6 +476,22 @@ function test_ListOfSupportedNonlinearOperators()
     return
 end
 
+function test_SPRAL()
+    if VERSION < v"1.9" || !haskey(ENV, "OMP_CANCELLATION")
+        return
+    end
+    model = Ipopt.Optimizer()
+    MOI.set(model, MOI.RawOptimizerAttribute("linear_solver"), "spral")
```
Comment (odow) [inline test/MOI_wrapper.jl:484]:
```suggestion
    MOI.set(model, MOI.RawOptimizerAttribute("linear_solver"), "spral")
```

**Comment by odow** [comment]
Looks like this worked. 

Last commit on `master` ran

image

this PR ran

image

**Review by odow — APPROVED** [review]

**Pull request merged**